src/base/projects/BranchSelector.svelte

```diff
@@ -101,14 +102,14 @@
     </div>
   <!-- If there is no branch listing available, show default branch name if commit is head and else show entire commit -->
   {:else if commit === project.head}
-    <div class="stat branch not-allowed">
+    <div data-cy="branch-name" class="stat branch not-allowed">
       {project.meta.defaultBranch}
     </div>
-    <div class="hash">
+    <div data-cy="hash" class="hash">
       {formatCommit(commit)}
     </div>
   {:else}
-    <div class="hash desktop">
+    <div data-cy="hash" class="hash desktop">
```

Comment:
A comment in general for the component test I think the best way is to use this `data-cy` attribute, since classes and tags can change but this attribute we can keep and is not scoped on a specific component, but is also valid for nested components.